index.bs

```diff
@@ -1081,6 +1115,104 @@ The <dfn attribute for="XRInputSource">targetRaySpace</dfn> attribute is an {{XR
 
 The {{gripSpace}} MUST be <code>null</code> if the input source isn't inherently trackable such as for input sources with a {{targetRayMode}} of {{XRTargetRayMode/gaze}} or {{XRTargetRayMode/screen}}.
+
+The <dfn attribute for="XRInputSource">gamepad</dfn> attribute is a {{Gamepad}} that describes the state of any buttons and axes on the input device. If the {{XRInputSource}} does not represent a device with either multiple buttons or at least one axis the {{gamepad}} attribute MUST be <code>null</code>. 
+</section>
+
+<section class="unstable">
+Gamepad API Integration {#gamepad-api-integration}
+-----------------------
+
+{{Gamepad}} instances returned by an {{XRInputSource}}'s {{XRInputSource/gamepad}} attribute behave as described by the <a href="https://w3c.github.io/gamepad/">Gamepad API</a>, with several additional behavioral restrictions:
+
+  - {{XRInputSource/gamepad}} MUST NOT be included in the array returned by {{navigator.getGamepads()}}.
+  - {{XRInputSource/gamepad}}'s {{Gamepad/index}} attribute MUST be <code>-1</code>.
+  - {{XRInputSource/gamepad}}'s {{Gamepad/connected}} attribute MUST be <code>true</code> until the {{XRInputSource}} is removed from the array returned by {{XRSession/getInputSources()}} or the {{XRSession}} is ended.
+
+ISSUE: Document {{Gamepad/id}} restrictions and formatting.
+</section>
+
+<section class="unstable">
+"xr-standard" Gamepad Mapping {#xr-standard-gamepad-mapping}
+-----------------------------
+
+The WebXR Device API introduces a new {{GamepadMappingType}} to describe the mapping of common XR controller devices.
+
+<pre class="idl">
+  enum GamepadMappingType {
+    "xr-standard",
+  };
+</pre>
+
+The <dfn enum-value for="GamepadMappingType">xr-standard</dfn> mapping indicates that the layout of the buttons and axes of the {{XRInputSource/gamepad}} corresponds as closely as possible to the tables below. In order to report a {{Gamepad/mapping}} of {{GamepadMappingType/xr-standard}} the device MUST report a {{XRInputSource/targetRayMode}} of {{XRTargetRayMode/tracked-pointer}}, MUST have at least one touchpad or joystick and MUST have at least one primary button, often a trigger, separate from the touchpad or joystick. If a device does not meet the requirements for the {{GamepadMappingType/xr-standard}} mapping it may still be exposed on an {{XRInputSource}} with the <code>""</code> (empty string) mapping.
```

Comment:
> If a device does not meet the requirements for the {{GamepadMappingType/xr-standard}} mapping it may still be exposed on an {{XRInputSource}} with the <code>""</code> (empty string) mapping.

I assume this is for compatibility with Gamepad?  What about "xr-unknown" instead?

Reply:
Yes, it's for gamepad compatability. I'm not sure there's a benefit to defining a new enum here for "not mapped".